### PR TITLE
Use custom_tags in http_connection_manager filter for sidecar proxies

### PR DESCRIPTION
**Pull request by jacob-delgado**
Partial fix for https://github.com/istio/istio/issues/13018

When the user has specified mesh wide configuration for proxies to use
custom_tags, propagate thos settings to http_connection_manager filter.

Depends on https://github.com/istio/api/pull/1369

### CODE REVIEW

**Comment by jacob-delgado** [comment]
I need to disregard gateways. As of now, their http connection manager is also modified. See https://github.com/istio/istio/blob/1c1f795db81e201758ec3541cce9fcdc1b7e3174/pilot/pkg/networking/core/v1alpha3/listener.go#L445

It's a risk to trust headers from external clients.

File: pilot/pkg/networking/core/v1alpha3/listener.go [hunk 1]
```diff
@@ -2009,6 +2008,54 @@ func buildTracingConfig() *http_conn.HttpConnectionManager_Tracing {
...
+				Type: &envoy_type_tracing_v2.CustomTag_Literal_{
+					Literal: &envoy_type_tracing_v2.CustomTag_Literal{
+						Value: c.Literal.Value,
+					},
+				},
+			}
+			tags = append(tags, env)
+		}
+	}
+	return tags
```
Comment (nrjpoddar) [inline pilot/pkg/networking/core/v1alpha3/listener.go:2058]: Sort the tags before returning. 

Use stable sort

File: pilot/pkg/networking/core/v1alpha3/listener_test.go [hunk 2]
```diff
@@ -1403,6 +1404,121 @@ func TestHttpProxyListener(t *testing.T) {
 	}
 }
 
+func TestHttpProxyListener_CustomTags(t *testing.T) {
+	var customTagsTest = []struct {
```
Comment (nrjpoddar) [inline pilot/pkg/networking/core/v1alpha3/listener_test.go:1408]: Add a test case for router and sidecar.

Comment (jacob-delgado) [reply pilot/pkg/networking/core/v1alpha3/listener_test.go:1408]: Added.

@nrjpoddar For gateways, do we want to add support for literals and env and only disregard request headers from external clients?

Right now it does not respect any custom_tag for gateways.

**Comment by akashjain971** [comment]
Please add support for customising tracing tags based on response headers as well.

**Comment by howardjohn** [comment]
It is

```
$ kubectl explain telemetry.spec.tracing.customTags.header
GROUP:      telemetry.istio.io
KIND:       Telemetry
VERSION:    v1

FIELD: header <Object>


DESCRIPTION:
    RequestHeader adds the value of an header from the request to each span.

FIELDS:
  defaultValue  <string>
    Optional.

  name  <string> -required-
    Name of the header from which to extract the tag value.
```

**Comment by howardjohn** [comment]
Sorry, I specific read your message to see if you said req or resp and could have sworn you said **request**.

Currently envoy only supports request headers

**Comment by akashjain971** [comment]
I see that envoy itself does not support response headers.
It has to come there before Istio can support.

**Pull request merged**